subsys/shell/Kconfig.backends

```diff
@@ -167,6 +167,92 @@ source "subsys/logging/Kconfig.template.log_config"
 
 endif #SHELL_BACKEND_RTT
 
+config SHELL_BACKEND_TELNET
+	bool "Enable TELNET backend."
```

Comment:
Add here also dependency to
* TCP
* IPv4
* IPv6

Currently it is possible to configure system so that both IPv4 and IPv6 are disabled, so we need to cover that here.